### DISCUSSION TITLE
Error After "source .zshrc"

### DISCUSSION THREAD

**Discussion by dmliu010** [marlonrichert/zsh-autocomplete#276] [Q&A] [answered]
Hello,

I am getting this error after running source on .zshrc. Problem is fixed after restarting the terminal, but would like to not have to do that every time I edit the .zshrc file.

`.autocomplete._main_complete:9: maximum nested function level reached; increase FUNCNEST?
.autocomplete._main_complete:unfunction:12: no such hash table element: compadd
.autocomplete._main_complete:unfunction:12: no such hash table element: compadd
.autocomplete._main_complete:unfunction:12: no such hash table element: compadd
.autocomplete._main_complete:unfunction:12: no such hash table element: compadd
.autocomplete._main_complete:unfunction:12: no such hash table element: compadd
.autocomplete._main_complete:unfunction:12: no such hash table element: compadd
.autocomplete._main_complete:unfunction:12: no such hash table element: compadd
.autocomplete._main_complete:unfunction:12: no such hash table element: compadd
.autocomplete._main_complete:unfunction:12: no such hash table element: compadd
.autocomplete._main_complete:unfunction:12: no such hash table element: compadd
_main_complete:unfunction:20: no such hash table element: compadd`

**Comment by marlonrichert** [comment, marked as answer]
As a rule, you should _not_ `source ~/.zshrc`. More info:
* [Weird things happen after typing `source ~/.zshrc`](https://github.com/romkatv/powerlevel10k/blob/master/README.md#weird-things-happen-after-typing-source-zshrc)
* [How do I reload the zshrc file?](https://github.com/ohmyzsh/ohmyzsh/wiki/FAQ#how-do-i-reload-the-zshrc-file)
* [How do I apply the changes to the .zshrc file after editing it?](https://unix.stackexchange.com/questions/532548/how-do-i-apply-the-changes-to-the-zshrc-file-after-editing-it/532591)
